raster-foundry/raster-foundry
# Use Binary Serialization for Mosaic Definition Cache

## Overview

On `develop` we currently have a big problem getting mosaic definitions from the cache. This caused some issues for large projects that have big mosaic definitions (one per scene in the project).

After a bunch of load testing, debugging, and tracing, the ultimate change ended up being to change the serialization for mosaic definition to binary serialization.

### Checklist


### Demo

Timings with circe serialization:
![image](https://user-images.githubusercontent.com/898060/66403864-5a9b7280-e9b5-11e9-860c-824c35c1fa62.png)

Timings on this branch:
![image](https://user-images.githubusercontent.com/898060/66403883-5ff8bd00-e9b5-11e9-89ae-8ee16020a12c.png)

So we go from > 1.5 _seconds_ to 11 or so _ms_

### Notes

I didn't change any of the other serializations because of an issue encountered serializing back to a case class with some of the other models. However, this one worked fine.

## Testing Instructions

- Re-assemble backsplash and api server
- Set [project](http://localhost:9091/projects/edit/0cd586b6-7ca7-4dac-a338-e89d6f0e1a48/sharing) to public
- Load http://localhost:8081/0cd586b6-7ca7-4dac-a338-e89d6f0e1a48/12/1192/1548/ in the browser
- Include any setup required, such as bundling scripts, restarting services, etc.
- Include test case, and expected output

Closes #5096
